### ISSUE TITLE
Type errors in v5

### ISSUE THREAD

**Issue by dzonatan** [MrWolfZ/ngrx-forms#155]
After upgrade to v5 I got all sorts of type errors.
It seems that all of these type errors are related to **array form controls** with **projection** functions of `setValue` and `validate`.

To keep things simple, below I have copy-pasted some of the examples from [documentation](https://ngrx-forms.readthedocs.io) that does have a type error in v5:

https://ngrx-forms.readthedocs.io/en/master/user-guide/updating-the-state/#setting-the-value
```ts
const array = createFormArrayState('array ID', ['']);
const updatedArray = setValue(['newValue'])(array); // <-- argument type error
```

https://ngrx-forms.readthedocs.io/en/master/user-guide/updating-the-state/#validating-the-value
```ts
const array = createFormArrayState<string>('array ID', ['']);
const updatedArray = validate<string[]>(value => value.length === 0 ? { missing: true } : {})(array); // <-- argument type error
```

https://ngrx-forms.readthedocs.io/en/master/user-guide/updating-the-state/#updating-groups
```ts
export interface MyFormValue {
  someTextInput: string;
  someCheckbox: boolean;
  nested: { someNumber: number };
  someNumbers: number[];
}

const updateMyFormGroup = updateGroup<MyFormValue>({
  someTextInput: validate(required),
  nested: updateGroup<MyFormValue['nested']>({
    someNumber: validate(required, greaterThanOrEqualTo(2)),
  }),
  someNumbers: validate(minLength(3)), // <-- type error
});
```

**Comment by MrWolfZ** [comment]
Which TypeScript version are you using? v5 requires TypeScript 3.4.x

**Comment by dzonatan** [comment]
Typescript `3.4.3`, Angular v8.
You can scaffold a new angular project (`ng new`) and copy-paste any of the example above and you should see type error.

**Comment by dzonatan** [comment]
After looking into this a bit more I can tell that all these type errors appears because of the `readonly` modifier which was [introduced in v5](https://github.com/MrWolfZ/ngrx-forms/commit/95a756b66b769bdb37fab15160176c77dc50d6e7). However, everything can be fixed by explicitly defining types. For example:
```ts
const array = createFormArrayState('array ID', ['']);

// type error
setValue(['newValue'])(array);

// works ok
setValue<readonly string[]>(['newValue'])(array);
setValue(['newValue'] as readonly string[])(array);
```

Same applies for other projection functions like `validate`, `updateArray`, ...
I don't know if this is intentional but it looks somewhat cumbersome.

**Comment by MrWolfZ** [comment]
I apologize for being so slow to respond to this issue. Thank you for pointing this issue out and for giving good examples for reproducing it. Having to do those casts was indeed not intended. I have just released version 5.0.3 which removes the `readonly` modifier from array state values. This should restore the original behavior.
